apps/emqx_mt/src/emqx_mt_app.erl

```diff
@@ -0,0 +1,17 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2024-2025 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+
+-module(emqx_mt_app).
+
+-behaviour(application).
+
+-export([start/2, stop/1]).
+
+start(_StartType, _StartArgs) ->
+    ok = emqx_mt_hookcb:register_hooks(),
+    emqx_mt_sup:start_link().
```

Comment:
I think the hook registration needs to happen after table creation to avoid some transient crashes:

```suggestion
    {ok, Sup} = emqx_mt_sup:start_link(),
    ok = emqx_mt_hookcb:register_hooks(),
    {ok, Sup}
```